### PR TITLE
23882 VAOS service GET appointment by id

### PR DESCRIPTION
**Pull request by cferris32**
## Description of change

This work is part of the ongoing VAOS service integration and adds the endpoint for retrieving a single appointment by id.

## Original issue(s)
https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/23882

## Things to know about this PR

This is one of many new vertical slices being built as part of the new VAOS service.

### CODE REVIEW

**Review by rileyanderson — DISMISSED** [review]

File: modules/vaos/spec/services/v2/appointment_service_spec.rb [hunk 1]
```diff
@@ -9,6 +9,8 @@
   let(:start_date) { Time.zone.parse('2020-06-02T07:00:00Z') }
   let(:end_date) { Time.zone.parse('2020-07-02T08:00:00Z') }
   let(:id) { '202006031600983000030800000000000000' }
+  let(:icn) { 543 }
```
Comment (grossb1) [inline modules/vaos/spec/services/v2/appointment_service_spec.rb:12]: See above re: ICN

File: modules/vaos/spec/services/v2/appointment_service_spec.rb [hunk 2]
```diff
@@ -32,4 +34,25 @@
       end
     end
   end
+
+  describe '#get_appointment' do
+    context 'with an appointment' do
+      it 'returns an appointment' do
+        VCR.use_cassette('vaos/v2/appointments/get_appointment', match_requests_on: %i[method uri]) do
+          response = subject.get_appointment(icn, appointment_id)
```
Comment (grossb1) [inline modules/vaos/spec/services/v2/appointment_service_spec.rb:42]: See above re: ICN

File: modules/vaos/spec/services/v2/appointment_service_spec.rb [hunk 3]
```diff
@@ -32,4 +34,25 @@
...
+          expect(response[:id]).to eq(id)
+        end
+      end
+    end
+
+    context 'when the upstream server returns a 500' do
+      it 'raises a backend exception' do
+        VCR.use_cassette('vaos/v2/appointments/get_appointment_500', match_requests_on: %i[method uri]) do
+          expect { subject.get_appointment(icn, appointment_id) }.to raise_error(
```
Comment (grossb1) [inline modules/vaos/spec/services/v2/appointment_service_spec.rb:51]: See above re: ICN

File: modules/vaos/app/docs/vaos/v2/vaos_v2.yaml [hunk 4]
```diff
@@ -93,6 +93,56 @@ paths:
           $ref: "#/components/responses/ServiceUnavailable"
         "504":
           $ref: "#/components/responses/GatewayTimeout"
+  "/patients/{icn}/appointments/{appointment-id}":
```
Comment (grossb1) [inline modules/vaos/app/docs/vaos/v2/vaos_v2.yaml:96]: I believe that ICN will be populated automatically in the service layer, we shouldn't need to pass it explicitly as a parameter.

File: spec/support/schemas/vaos/v2/appointment.json [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+          "$ref": "#/components/schemas/AppointmentKind"
+        },
+        "status": {
+          "$ref": "#/components/schemas/AppointmentStatus"
+        },
+        "serviceType": {
+          "description": "the care type for the appointment",
+          "type": "string"
+        },
+        "patientIcn": {
```
Comment (grossb1) [inline spec/support/schemas/vaos/v2/appointment.json:30]: This can be removed

**Review by grossb1 — APPROVED** [review]

**Pull request merged**